arangod/Utils/SingleCollectionTransaction.h

```diff
@@ -38,11 +38,18 @@ class SingleCollectionTransaction final : public transaction::Methods {
 
  public:
   //////////////////////////////////////////////////////////////////////////////
-  /// @brief create the transaction, using a collection id
+  /// @brief create the transaction, using a collection
   //////////////////////////////////////////////////////////////////////////////
 
   SingleCollectionTransaction(std::shared_ptr<transaction::Context> const&,
-                              TRI_voc_cid_t, AccessMode::Type);
+                              LogicalCollection const*, AccessMode::Type);
```

Comment:
it is not advisable to pass raw pointers if a value is actually expected to always be non-null
otherwise it requires use of extra null checks, a reference might be more beneficial

Reply:
agreed, but now that I changed about 90 different files I would like to avoid going back and doing it again